### ISSUE TITLE
Sometimes, missing HTTP Content-Length header

### ISSUE THREAD

**Issue by danielweck** [divshot/superstatic#169]
In our app, we rely on the `Content-Length` HTTP header, which appears to be missing in some cases when requesting assets on DivShot.io:

Examples:

`curl "http://assets.divshot.io/superstatic/background.jpg" -I`
=>
```
HTTP/1.1 200 OK
Date: Tue, 02 Jun 2015 22:49:35 GMT
Content-Type: image/jpeg
Connection: keep-alive
Cache-Control: public, max-age=31536000
Content-Encoding: identity
Server: NetDNA-cache/2.2
X-Cache: HIT
```

---

`curl "http://readium-cloudreader.divshot.io/index.html" -I`
=>
```
HTTP/1.1 200 OK
Cache-Control: public, max-age=300
Content-Encoding: identity
Content-Type: text/html; charset=utf-8
Date: Tue, 02 Jun 2015 22:47:54 GMT
Server: nginx/1.4.3
Vary: Accept-Encoding
X-Accel-Expires: 31536000
Connection: keep-alive
```

---

`curl "http://gulpfiction.divshot.io/img/logo2.gif" -I`
=>
```
HTTP/1.1 200 OK
Date: Tue, 02 Jun 2015 22:55:23 GMT
Content-Type: image/gif
Connection: keep-alive
Cache-Control: public, max-age=300
Content-Encoding: identity
Server: NetDNA-cache/2.2
X-Cache: HIT
```

---

Yet, this one works:

`curl "http://readium-cloudreader.divshot.io/favicon.ico" -I`
=>
```
HTTP/1.1 200 OK
Cache-Control: public, max-age=300
Content-Length: 6518
Content-Type: image/x-icon
Date: Tue, 02 Jun 2015 22:47:26 GMT
ETag: "XiumV/vxTaN6sirlG7eQLA=="
Server: nginx/1.4.3
Vary: Accept-Encoding
X-Accel-Expires: 31536000
Connection: keep-alive
```

Any idea why?

**Comment by scottcorgan** [comment]
This seems to be fixed on your end. Going to close. 

Reopen if you see it again.

**Comment by danielweck** [comment]
@scottcorgan thanks for monitoring this issue. However, the problem is *not fixed at your end*. I do not have sufficient credentials to re-open this issue, so I will leave that to you :)

We implemented a workaround at our end ; specifically for DivShot.io ; but this is (very) suboptimal as it forces us to load entire zip files in memory (binary Blobs) instead of fetching them via HTTP byte-range protocol (to access individual resources within the compressed archive).

The `Content-Length` header is pretty much essential, as you can see in:
https://github.com/gildas-lormeau/zip.js/blob/master/WebContent/zip-ext.js
(search for "Content-Length" in the JS source)

Here's a CURL request log (missing HTTP `Content-Length` header):
```
$  curl "http://development.readium.divshot.io/epub_content/internal_link.epub" -I
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Headers: Range, Content-Type, Origin, X-Requested-With, Accept, Accept-Language, Content-Language
Access-Control-Allow-Methods: GET, HEAD, POST
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: Accept-Ranges, Content-Encoding, Content-Type, Content-Length, Content-Range, Content-Language, Cache-Control, Expires, Last-Modified, Pragma
Cache-Control: public, max-age=31536000
Content-Encoding: identity
Content-Type: application/epub+zip
Date: Wed, 23 Sep 2015 12:41:36 GMT
X-Accel-Expires: 31536000
X-Robots-Tag: noindex
```

**Comment by danielweck** [comment]
Hello @scottcorgan just a quick "bump": as I mentioned in my previous message, the problem persists (missing `Content-Length` HTTP header), so this issue should not be closed, right?

We currently implement a workaround, but this forces us to download potentially huge ZIP files + store them in memory (Blob URI), instead of streaming chunks of archives over HTTP (byte range requests) in order to extract specific files from the ZIP catalog.

Thanks! :)

**Comment by mbleigh** [comment]
Hey @danielweck this bug is being addressed in Superstatic 4.0 work, which will be rolled out to Firebase Hosting when it's ready. We are not planning to address this for Divshot hosting since Divshot will be shutting down services in the near future and we're focusing on improving Firebase Hosting.

Thanks!
